pkg/storage/ingest/reader.go

```diff
@@ -1076,22 +1070,84 @@ func (r *concurrentFetchers) runFetchers(ctx context.Context, startOffset int64)
 	}
 }
 
-func logCompletedFetch(logger log.Logger, f fetchResult, fetchStartTime time.Time, attempt int, w fetchWant) {
+// handleKafkaFetchErr handles all the errors listed in the franz-go documentation as possible errors when fetching records.
+// For most of them we just apply a backoff. They are listed here so we can be explicit in what we're handling and how.
+// handleKafkaFetchErr returns an adjusted fetchWant in case the error indicated we were consuming not yet produced records or records already deleted due to retention.
```

Comment:
```suggestion
// it may also return an adjusted fetchWant in case the error indicated, we were consuming not yet produced records or records already deleted due to retention.
```